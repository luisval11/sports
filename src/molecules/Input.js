import { Form } from "react-bootstrap";
import LabelI from "../atoms/label/Label";

const Input = (props) => {
	return (
		<Form.Group
			controlId={props.index}
			key={props.index}
			className={props.className}
		>
			{props.label && <LabelI text={props.label} />}
			<Form.Control
				id={props.id}
				as={props.as}
				type={props.type}
				name={props.name}
				placeholder={props.placeholder}
				min={props.min}
				max={props.max}
				value={props.value}
				autoComplete={props.autoComplete}
				onChange={(event) => {
					props.handleInput && props.handleInput(event);
				}}
			/>
		</Form.Group>
	);
};
export default Input;
