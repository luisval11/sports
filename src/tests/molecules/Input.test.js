import { mount, shallow } from "enzyme";
import Input from "../../molecules/Input";
import { CardTitle } from "react-bootstrap";

describe("Test on Input", () => {

	test("should match initial snapshot", () => {
        let wrapper = shallow(
            <Input
                className="mt-3"
                type="text"
                label="Home Team"
                name="homeTeam"
                placeholder="Spain"
                autoComplete="true"
            />
        );
		expect(wrapper).toMatchSnapshot();
	});

	test("Should have a label of as 'Home Team'", () => {
        let wrapper = mount(
            <Input
                className="mt-3"
                type="text"
                label="Home Team"
                name="homeTeam"
                placeholder="Spain"
                autoComplete="true"
            />
        );
        expect(wrapper.find('.form-label').text().trim()).toBe('Home Team');
	});
});
