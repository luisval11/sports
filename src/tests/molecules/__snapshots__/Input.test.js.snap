// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on Input should match initial snapshot 1`] = `
<FormGroup
  className="mt-3"
>
  <LabelI
    text="Home Team"
  />
  <FormControl
    autoComplete="true"
    name="homeTeam"
    onChange={[Function]}
    placeholder="Spain"
    type="text"
  />
</FormGroup>
`;
